lib/manager.coffee

```diff
@@ -36,6 +36,7 @@ doDownload = (deviceType, version) ->
 		cache.getImageWritableStream(deviceType, version)
 		.then (cacheStream) ->
 			pass.pipe(cacheStream)
+			pass.on('end', cacheStream.persistCache)
```

Comment:
Should we wait until the `cacheStream.persistCache` promise is resolved before resolving this promise?

Reply:
That would be tidy, but we can't I think.

The promise resolves with the stream itself, and needs to resolve as soon as that starts, so we can start streaming it onwards, and show progress while the download runs. If we wait until the stream ends and the cache is persisted then we don't get that streaming goodness.